code/modules/station_goals/bsa.dm

```diff
@@ -380,41 +394,32 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/power/bsa/full)
 		if("fire")
 			fire(usr)
 			. = TRUE
-		if("recalibrate")
-			calibrate(usr)
+		if("set_target")
+			var/datum/component/gps/target = locate(params["target"])
+			target_ref = WEAKREF(target)
```

Comment:
```suggestion
		if("set_target")
			var/datum/component/gps/target = locate(params["chosen_target"])
			target_ref = WEAKREF(target)
```
Even if I requested "target" not being a list anymore and putting into different list keys, this can make confusion. It's better to have a different key name. (The change request goes in `js` file too)

Reply:
done